1.31.0

matteobruni opened the pull request:
# tsParticles 1.31.0

## New Features

- Added zIndex options, closes #979
- Added particles groups to options

## Changes

- Infection is now a plugin, and it's included and is not included in the slim version
- Changed particles move distance options to an horizontal/vertical object, single number still valid

matteobruni committed:
chore(main): migrating from Utils static objects to public functions

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/0c67162ef385341c0adbcbbdd885249ea12fcc6c/matteobruni/tsparticles/f17972005444aa7635d8e52b694f71cb26ec47a2/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) your code in 6.439 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/0c67162ef385341c0adbcbbdd885249ea12fcc6c/matteobruni/tsparticles/f17972005444aa7635d8e52b694f71cb26ec47a2/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

matteobruni committed:
Merge branch 'main' into dev

matteobruni committed:
build: updates after merge

matteobruni committed:
build: code style fixes

matteobruni committed:
build: code style fixes

dependabot[bot] committed:
build(deps-dev): bump @types/webpack-env from 1.16.1 to 1.16.2

Bumps [@types/webpack-env](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/webpack-env) from 1.16.1 to 1.16.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/webpack-env)

---
updated-dependencies:
- dependency-name: "@types/webpack-env"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint from 7.29.0 to 7.30.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.29.0 to 7.30.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.29.0...v7.30.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 15.14.0 to 16.0.0

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 15.14.0 to 16.0.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.41.1 to 5.42.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.41.1 to 5.42.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.41.1...v5.42.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1758 from matteobruni/dependabot/npm_and_yarn/dev/types/webpack-env-1.16.2

build(deps-dev): bump @types/webpack-env from 1.16.1 to 1.16.2

matteobruni committed:
Merge pull request #1759 from matteobruni/dependabot/npm_and_yarn/dev/eslint-7.30.0

build(deps-dev): bump eslint from 7.29.0 to 7.30.0

matteobruni committed:
Merge pull request #1760 from matteobruni/dependabot/npm_and_yarn/dev/types/node-16.0.0

build(deps-dev): bump @types/node from 15.14.0 to 16.0.0

matteobruni committed:
Merge pull request #1761 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.42.0

build(deps-dev): bump webpack from 5.41.1 to 5.42.0

dependabot[bot] committed:
build(deps-dev): bump mocha from 9.0.1 to 9.0.2

Bumps [mocha](https://github.com/mochajs/mocha) from 9.0.1 to 9.0.2.
- [Release notes](https://github.com/mochajs/mocha/releases)
- [Changelog](https://github.com/mochajs/mocha/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mochajs/mocha/compare/v9.0.1...v9.0.2)

---
updated-dependencies:
- dependency-name: mocha
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @vue/compiler-sfc from 3.1.3 to 3.1.4

Bumps [@vue/compiler-sfc](https://github.com/vuejs/vue-next/tree/HEAD/packages/compiler-sfc) from 3.1.3 to 3.1.4.
- [Release notes](https://github.com/vuejs/vue-next/releases)
- [Changelog](https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vue-next/commits/v3.1.4/packages/compiler-sfc)

---
updated-dependencies:
- dependency-name: "@vue/compiler-sfc"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
build: code style fixes

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
Merge pull request #1762 from matteobruni/dependabot/npm_and_yarn/dev/mocha-9.0.2

build(deps-dev): bump mocha from 9.0.1 to 9.0.2

matteobruni committed:
Merge pull request #1763 from matteobruni/dependabot/npm_and_yarn/dev/vue/compiler-sfc-3.1.4

build(deps-dev): bump @vue/compiler-sfc from 3.1.3 to 3.1.4

matteobruni committed:
build: updated some deps

matteobruni committed:
build: disabled typesVersions, still buggy

matteobruni committed:
build: updates

matteobruni committed:
Merge branch 'main' into dev

matteobruni committed:
chore(main): changed particle instance life properties

matteobruni committed:
chore(main): moved infection to a plugin like in v2
chore(main): migration to v2 plugin system (updaters and interactors are also plugins)

matteobruni committed:
chore(main): all plugins have the new structure

matteobruni committed:
build: make things build for now, editor needs some fixes

matteobruni committed:
feat: zIndex, closes #979
feat: particles groups

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 4.28.1 to 4.28.2

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 4.28.1 to 4.28.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.28.2/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump rxjs from 7.1.0 to 7.2.0

Bumps [rxjs](https://github.com/reactivex/rxjs) from 7.1.0 to 7.2.0.
- [Release notes](https://github.com/reactivex/rxjs/releases)
- [Changelog](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/reactivex/rxjs/compare/7.1.0...7.2.0)

---
updated-dependencies:
- dependency-name: rxjs
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.28.1 to 4.28.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.28.2/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.42.0 to 5.42.1

Bumps [webpack](https://github.com/webpack/webpack) from 5.42.0 to 5.42.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.42.0...v5.42.1)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1765 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/parser-4.28.2

build(deps-dev): bump @typescript-eslint/parser from 4.28.1 to 4.28.2

matteobruni committed:
Merge pull request #1766 from matteobruni/dependabot/npm_and_yarn/dev/rxjs-7.2.0

build(deps): bump rxjs from 7.1.0 to 7.2.0

matteobruni committed:
Merge pull request #1767 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/eslint-plugin-4.28.2

build(deps-dev): bump @typescript-eslint/eslint-plugin from 4.28.1 to 4.28.2

matteobruni committed:
Merge pull request #1768 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.42.1

build(deps-dev): bump webpack from 5.42.0 to 5.42.1

matteobruni committed:
fix: fixed deletion when surpassing limit and z index is enabled

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.42.1 to 5.43.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.42.1 to 5.43.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.42.1...v5.43.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/mocha from 8.2.2 to 8.2.3

Bumps [@types/mocha](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/mocha) from 8.2.2 to 8.2.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/mocha)

---
updated-dependencies:
- dependency-name: "@types/mocha"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump husky from 7.0.0 to 7.0.1

Bumps [husky](https://github.com/typicode/husky) from 7.0.0 to 7.0.1.
- [Release notes](https://github.com/typicode/husky/releases)
- [Commits](https://github.com/typicode/husky/compare/v7.0.0...v7.0.1)

---
updated-dependencies:
- dependency-name: husky
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-vue from 7.12.1 to 7.13.0

Bumps [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) from 7.12.1 to 7.13.0.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v7.12.1...v7.13.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-vue
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/chai from 4.2.19 to 4.2.20

Bumps [@types/chai](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/chai) from 4.2.19 to 4.2.20.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/chai)

---
updated-dependencies:
- dependency-name: "@types/chai"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1770 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.43.0

build(deps-dev): bump webpack from 5.42.1 to 5.43.0

matteobruni committed:
Merge pull request #1771 from matteobruni/dependabot/npm_and_yarn/dev/types/mocha-8.2.3

build(deps-dev): bump @types/mocha from 8.2.2 to 8.2.3

matteobruni committed:
Merge pull request #1772 from matteobruni/dependabot/npm_and_yarn/dev/husky-7.0.1

build(deps-dev): bump husky from 7.0.0 to 7.0.1

matteobruni committed:
Merge pull request #1773 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-vue-7.13.0

build(deps-dev): bump eslint-plugin-vue from 7.12.1 to 7.13.0

matteobruni committed:
Merge pull request #1774 from matteobruni/dependabot/npm_and_yarn/dev/types/chai-4.2.20

build(deps-dev): bump @types/chai from 4.2.19 to 4.2.20

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.52.7 to 2.52.8

Bumps [rollup](https://github.com/rollup/rollup) from 2.52.7 to 2.52.8.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.52.7...v2.52.8)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/jquery from 3.5.5 to 3.5.6

Bumps [@types/jquery](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jquery) from 3.5.5 to 3.5.6.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jquery)

---
updated-dependencies:
- dependency-name: "@types/jquery"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1775 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.52.8

build(deps-dev): bump rollup from 2.52.7 to 2.52.8

matteobruni committed:
Merge pull request #1776 from matteobruni/dependabot/npm_and_yarn/dev/types/jquery-3.5.6

build(deps-dev): bump @types/jquery from 3.5.5 to 3.5.6

matteobruni committed:
build: setup firebase for checking PR deploys

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
build: preparing a new vanilla demo to be used in firebase

matteobruni committed:
build: preparing a new vanilla demo to be used in firebase

matteobruni committed:
build: updated gitignore

matteobruni committed:
build: updated gitignore

matteobruni committed:
build: updated new vanilla demo website

matteobruni committed:
build: fixed local tsparticles script in new vanilla demo website

dependabot[bot] committed:
build(deps): bump sass from 1.35.1 to 1.35.2

Bumps [sass](https://github.com/sass/dart-sass) from 1.35.1 to 1.35.2.
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.35.1...1.35.2)

---
updated-dependencies:
- dependency-name: sass
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/react-dom from 17.0.8 to 17.0.9

Bumps [@types/react-dom](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react-dom) from 17.0.8 to 17.0.9.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react-dom)

---
updated-dependencies:
- dependency-name: "@types/react-dom"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/react from 17.0.13 to 17.0.14

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 17.0.13 to 17.0.14.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 16.0.0 to 16.0.1

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.0.0 to 16.0.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1780 from matteobruni/dependabot/npm_and_yarn/dev/types/node-16.0.1

build(deps-dev): bump @types/node from 16.0.0 to 16.0.1

matteobruni committed:
Merge pull request #1777 from matteobruni/dependabot/npm_and_yarn/dev/sass-1.35.2

build(deps): bump sass from 1.35.1 to 1.35.2

matteobruni committed:
Merge pull request #1779 from matteobruni/dependabot/npm_and_yarn/dev/types/react-17.0.14

build(deps-dev): bump @types/react from 17.0.13 to 17.0.14

matteobruni committed:
Merge pull request #1778 from matteobruni/dependabot/npm_and_yarn/dev/types/react-dom-17.0.9

build(deps-dev): bump @types/react-dom from 17.0.8 to 17.0.9

matteobruni committed:
build: moved firebase to node.js CI

matteobruni committed:
Merge branch 'dev' of https://github.com/matteobruni/tsparticles into dev

matteobruni committed:
build: updated component readmes for the peer dependency

matteobruni committed:
build: updated Node.js CI for firebase publish

matteobruni committed:
build: updated Node.js CI for firebase publish

matteobruni committed:
build: updated Node.js CI for firebase publish

matteobruni committed:
build: updated Node.js CI for firebase publish

matteobruni committed:
build: updated Node.js CI for firebase publish

matteobruni committed:
build: updated Node.js CI for firebase publish

matteobruni committed:
build: updated Node.js CI for firebase publish

matteobruni committed:
build: improving new vanilla demo website for firebase

matteobruni committed:
build: improving new vanilla demo website for firebase

matteobruni committed:
build: improving new vanilla demo website for firebase

dependabot[bot] committed:
build(deps-dev): bump @types/chai from 4.2.20 to 4.2.21

Bumps [@types/chai](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/chai) from 4.2.20 to 4.2.21.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/chai)

---
updated-dependencies:
- dependency-name: "@types/chai"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.43.0 to 5.44.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.43.0 to 5.44.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.43.0...v5.44.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 16.0.1 to 16.0.3

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.0.1 to 16.0.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump svelte-check from 2.2.0 to 2.2.1

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 2.2.0 to 2.2.1.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/compare/svelte-check-2.2.0...svelte-check-2.2.1)

---
updated-dependencies:
- dependency-name: svelte-check
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1781 from matteobruni/dependabot/npm_and_yarn/dev/types/chai-4.2.21

build(deps-dev): bump @types/chai from 4.2.20 to 4.2.21

matteobruni committed:
Merge pull request #1782 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.44.0

build(deps-dev): bump webpack from 5.43.0 to 5.44.0

matteobruni committed:
Merge pull request #1783 from matteobruni/dependabot/npm_and_yarn/dev/types/node-16.0.3

build(deps-dev): bump @types/node from 16.0.1 to 16.0.3

matteobruni committed:
Merge pull request #1784 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-2.2.1

build(deps-dev): bump svelte-check from 2.2.0 to 2.2.1

matteobruni committed:
build: moved editor and engine out of core folder

matteobruni committed:
build: fixed lerna.json

matteobruni committed:
build: fixed lerna.json

matteobruni committed:
build: fixed editor for new infection plugin

matteobruni committed:
build: fixed editor for new infection plugin

matteobruni committed:
build: simplified emitter needs plugin

matteobruni committed:
build: simplified needs plugin for absorbers

matteobruni committed:
build: fix old load

matteobruni committed:
build: updated shapes to new files structure without dist folder like the main package

matteobruni committed:
build: updated shapes to new files structure without dist folder like the main package

matteobruni committed:
build: updated shapes to new files structure without dist folder like the main package

matteobruni committed:
build: updated presets to new file structure without dist output folder

matteobruni committed:
feat: changed particles move distance to an horizontal/vertical object, single number still valid
build: fixed some demos
build: updated editor

matteobruni committed:
build: updated some demos

matteobruni committed:
chore(release): published new version

 - ng-particles@2.14.0
 - inferno-particles@1.14.0
 - jquery-particles@1.31.0
 - preact-particles@1.31.0
 - react-tsparticles@1.31.0
 - riot-particles@1.5.0
 - solid-particles@1.14.0
 - svelte-particles@1.14.0
 - particles.vue@2.14.0
 - particles.vue3@1.14.0
 - web-particles@1.4.0
 - ng-particles-demo@1.16.0
 - editor-demo@1.16.0
 - imports-ts-latest@1.2.0
 - imports-ts34@1.2.0
 - imports-ts38@1.2.0
 - imports-ts40@1.2.0
 - inferno-particles-demo@1.14.0
 - jquery-particles-demo@1.16.0
 - preact-particles-demo@1.16.0
 - react-tsparticles-demo@1.16.0
 - riot-particles-demo@1.5.0
 - solid-particles-demo@1.14.0
 - svelte-demo@1.14.0
 - tsparticles-demo@1.16.0
 - tsparticles-demo-new@1.16.0
 - particles.vue-demo@1.16.0
 - particles.vue3-demo@1.14.0
 - web-particles-demo@1.16.0
 - tsparticles-editor@1.31.0
 - tsparticles@1.31.0
 - tsparticles-preset-big-circles@1.16.0
 - tsparticles-preset-bubbles@1.16.0
 - tsparticles-preset-confetti@1.16.0
 - tsparticles-preset-fire@1.16.0
 - tsparticles-preset-firefly@1.16.0
 - tsparticles-preset-fireworks@1.16.0
 - tsparticles-preset-fountain@1.16.0
 - tsparticles-preset-links@1.16.0
 - tsparticles-preset-sea-anemone@1.16.0
 - tsparticles-preset-snow@1.16.0
 - tsparticles-preset-stars@1.16.0
 - tsparticles-shape-bubble@1.16.0
 - tsparticles-shape-heart@1.16.0
 - tsparticles-shape-multiline-text@1.16.0
 - tsparticles-shape-rounded-rect@1.16.0
 - tsparticles-shape-spiral@1.16.0
 - cra-template-particles-typescript@1.14.0
 - cra-template-particles@1.14.0
 - tsparticles-website@1.31.0

matteobruni committed:
build: website update
